tests/test_api.rs

```diff
@@ -2140,6 +2147,21 @@ fn snapshot_creator() {
       let result = script.run(scope).unwrap();
       let true_val = v8::Boolean::new(scope, true).into();
       assert!(result.same_value(true_val));
+
+      let isolate_data = unsafe {
+        isolate
```

Comment:
can't borrow isolate here, unsure how to test or if this design even works with the lifetimes